exporter/opentelemetry-exporter-prometheus/src/opentelemetry/exporter/prometheus/__init__.py

```diff
@@ -159,96 +162,153 @@ def collect(self) -> None:
+                            "%".join(label_keys),
+                            metric.unit,
+                        ]
+                    )
                 )
-            metric_family_id_metric_family[metric_family_id].add_metric(
-                labels=label_values, value=metric.point.value
-            )
-        elif isinstance(metric.point, Gauge):
-
-            metric_family_id = "|".join(
-                [metric_family_id, GaugeMetricFamily.__name__]
-            )
-
-            if metric_family_id not in metric_family_id_metric_family.keys():
-                metric_family_id_metric_family[
-                    metric_family_id
-                ] = GaugeMetricFamily(
-                    name=metric_name,
-                    documentation=description,
-                    labels=label_keys,
-                    unit=metric.unit,
-                )
-            metric_family_id_metric_family[metric_family_id].add_metric(
-                labels=label_values, value=metric.point.value
-            )
-        elif isinstance(metric.point, Histogram):
-
-            metric_family_id = "|".join(
-                [metric_family_id, HistogramMetricFamily.__name__]
-            )
 
-            if metric_family_id not in metric_family_id_metric_family.keys():
-                metric_family_id_metric_family[
-                    metric_family_id
-                ] = HistogramMetricFamily(
-                    name=metric_name,
-                    documentation=description,
-                    labels=label_keys,
-                    unit=metric.unit,
-                )
-            metric_family_id_metric_family[metric_family_id].add_metric(
-                labels=label_values,
-                buckets=_convert_buckets(metric),
-                sum_value=metric.point.sum,
-            )
-        else:
-            _logger.warning("Unsupported metric type. %s", type(metric.point))
+            for pre_metric_family_id, label_values, value in zip(
+                pre_metric_family_ids, label_valuess, values
+            ):
+                if isinstance(metric.data, Sum):
+
+                    metric_family_id = "|".join(
+                        [pre_metric_family_id, CounterMetricFamily.__name__]
+                    )
+
+                    if (
+                        metric_family_id
+                        not in metric_family_id_metric_family.keys()
+                    ):
+                        metric_family_id_metric_family[
+                            metric_family_id
+                        ] = CounterMetricFamily(
+                            name=metric_name,
+                            documentation=description,
+                            labels=label_keys,
+                            unit=metric.unit,
+                        )
+                    metric_family_id_metric_family[
+                        metric_family_id
+                    ].add_metric(labels=label_values, value=value)
```

Comment:
nit This is repeated a few times, can it be move to after the else statement below?

```py
if metric_family_id:
  metric_family_id_metric_family[
    metric_family_id
  ].add_metric(labels=label_values, value=value)
```

Reply:
It won't save us any code, actually it would require having another `if/else` because we don't do use the same parameters for `add_metric` when the metric is a histogram. :shrug: